quickwit-search/src/filters.rs

```diff
@@ -75,18 +110,23 @@ impl TimestampFilterBuilder {
             .schema()
             .get_field_entry(self.timestamp_field);
 
-        let expected_type = Type::I64;
         let field_schema_type = field_entry.field_type().value_type();
-        if expected_type != field_schema_type {
-            return Err(TantivyError::SchemaError(format!(
-                "Field {:?} is of type {:?}!={:?}",
-                field_entry.name(),
-                expected_type,
-                field_schema_type
-            )));
-        }
+        let timestamp_field_reader = match field_entry.field_type().value_type() {
+            Type::I64 => {
+                GenericFastFieldReader::I64(segment_reader.fast_fields().i64(self.timestamp_field)?)
+            }
+            Type::Date => GenericFastFieldReader::Date(
+                segment_reader.fast_fields().date(self.timestamp_field)?,
+            ),
+            _ => {
+                return Err(TantivyError::SchemaError(format!(
+                    "Field {:?} is of type {:?}, expected I64 or Date type.",
```

Comment:
"Failed to build timestamp filter for field `{}`: expected i64 or Date type, got {}. "